### PR TITLE
clean up quantum.c

### PR DESCRIPTION
**Pull request by yanfali**
functional equivalence, less code. I've tested this on an AN-C. Need to test it on a split with RGB.

## Description
Remove common boilerplate, migrate code into a different functionally equivalent switch while maintaining backwards compatibility

 - hoist the keypress detection
 - remove nesting
 - commonality among ifdefs

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/quantum.c [hunk 1]
```diff
@@ -277,258 +277,173 @@ bool process_record_quantum(keyrecord_t *record) {
     }
 
     // Shift / paren setup
```
Comment (fauxpark) [inline quantum/quantum.c]:
```suggestion
```

Relic of Space Cadet before it was pulled into process_keycode/

**Comment by drashna** [comment]
Tested on corne (without split common), and kyria, both seem to work just fine.

**Comment by yanfali** [comment]
Should we merge?

File: quantum/quantum.c [hunk 2]
```diff
@@ -276,259 +276,173 @@ bool process_record_quantum(keyrecord_t *record) {
...
                 } else {
                     rgblight_mode(RGBLIGHT_MODE_BREATHING);
                 }
-            }
-#    endif
-            return false;
-        case RGB_MODE_RAINBOW:
-#    ifdef RGBLIGHT_EFFECT_RAINBOW_MOOD
-            if (record->event.pressed) {
-                if ((RGBLIGHT_MODE_RAINBOW_MOOD <= rgblight_get_mode()) && (rgblight_get_mode() < RGBLIGHT_MODE_RAINBOW_MOOD_end)) {
```
Comment (zvecr) [inline quantum/quantum.c]: should this block not still exist, otherwise the `RGB_MODE_RAINBOW` keycode will stop working?

Comment (yanfali) [reply quantum/quantum.c]: oops

**Pull request merged**